gatsby-node.js

```diff
@@ -1,23 +1,98 @@
-// eslint-disable-next-line @typescript-eslint/no-var-requires
 const path = require(`path`);
+const fetch = require('node-fetch');
+const xmlParser = require('fast-xml-parser');
 
-const BLOG_POST_TEMPLATE_PATH = path.resolve(`src/templates/blog-post.tsx`);
-const CLIENT_TEMPLATE_PATH = path.resolve(`src/templates/client-details.tsx`);
+const BLOG_POST_TEMPLATE_PATH = path.resolve('src/templates/blog-post.tsx');
+const CLIENT_TEMPLATE_PATH = path.resolve('src/templates/client-details.tsx');
+const CAREER_TEMPLATE_PATH = path.resolve('src/templates/career.tsx');
+const CAREER_DETAILS_TEMPLATE_PATH = path.resolve(
+  'src/templates/career-details.tsx',
+);
 
-exports.createPages = async ({ actions, graphql, reporter }) => {
+const PERSONIO_JOBS_URL = 'https://satellytes.jobs.personio.de/xml';
+const PERSONIO_SHORT_DESCRIPTION_NAME = 'Kurzbeschreibung';
+
+exports.createPages = async (createPagesArgs) => {
+  await createCareerPages(createPagesArgs);
+  await createBlogPages(createPagesArgs);
+  // await createClientPages(createPagesArgs);
+};
+
+const createCareerPages = async ({ actions }) => {
   const { createPage } = actions;
 
-  const clientPages = await graphql(`
-    {
-      allClientsJson {
-        nodes {
-          path
-        }
-      }
+  const jobsXmlResponse = await fetch(PERSONIO_JOBS_URL);
+  const jobsXml = await jobsXmlResponse.text();
+  const jobsParse = xmlParser.parse(jobsXml);
+  const positions = jobsParse['workzag-jobs'].position;
+
+  const slugifyPositionName = (name) => {
+    return name
+      .toLowerCase()
+      .replace(/ /g, '-')
+      .replace(/-/g, '-')
+      .replace(/\//g, '-')
+      .replace(/\(/g, '')
+      .replace(/\)/g, '');
+  };
+
+  positions.forEach((position) => {
+    position.satellytesPath = `/career/${position.id}-${slugifyPositionName(
+      position.name,
+    )}`;
+
+    // we need to normalize the jobDescription to an array, otherwise gatsby
+    // complains about different types for the same variable
+    position.jobDescriptions.jobDescription = Array.isArray(
+      position.jobDescriptions.jobDescription,
+    )
+      ? position.jobDescriptions.jobDescription
+      : [position.jobDescriptions.jobDescription];
```

Comment:
The library either emits an object or an array depending on the node count being `>=2` that's why gatsby complains and why we need to normalize the value into an array.

![Screenshot 2021-03-25 at 07 21 07](https://user-images.githubusercontent.com/1701755/112427933-b63cf580-8d3a-11eb-9c40-84a14ca9dc81.png)

There is an `arrayMode` to always output and array for every node. I don't know if this would make it better here though. It works so I'm good with that.

I would adjust the comment though. It's not a problem from gatsby, it's the data structure.

You are using the ternary operator and I think this does not help reading that part here, why not this explicit beauty ?

```
// ensure given value is an array
if(!Array.isArray(position.jobDescriptions.jobDescription)) {
	position.jobDescriptions.jobDescription = [position.jobDescriptions.jobDescription];
}
```